tests/data/tavern/api_get_files_format.yaml

```diff
@@ -0,0 +1,107 @@
+---
+test_name: Check get_files format (json output)
+
+strict:
+  - json:on
+
+stages:
+  - name: get_files format (json)
+    request:
+      url: "http://{daemon_host}:{daemon_port}/sabnzbd/api"
+      method: GET
+      params:
+        mode: get_files
+        apikey: "{daemon_apikey}"
+        output: json
+        value: "{nzo_id}"
+    response:
+      status_code: 200
+      headers:
+        content-type: !re_match "application/json"
+        content-type: !re_search "charset=(UTF|utf)-8"
+        pragma: "no-cache"
+        access-control-allow-origin: "*"
+      json:
+        files: !anylist
+          filename: !anystr
+          mbleft: !anyfloat
+          mb: !anyfloat
+          bytes: !anyfloat
+          age: !anystr
+          nzf_id: !re_match "SABnzbd_nzf_.*"
+          status: !re_match "(finished|active|queued)"
+
+
+---
+test_name: Check get_files format (text output)
+
+strict:
+  - json:on
+
+stages:
+  - name: get_files format (text)
+    request:
+      url: "http://{daemon_host}:{daemon_port}/sabnzbd/api"
+      method: GET
+      params:
+        mode: get_files
+        apikey: "{daemon_apikey}"
+        output: text
+        value: "{nzo_id}"
+    response:
+      status_code: 200
+      headers:
+        content-type: !re_match "text/plain"
+        content-type: !re_search "charset=(UTF|utf)-8"
+        pragma: "no-cache"
+        access-control-allow-origin: "*"
+      verify_response_with:
+        function: tavern.testutils.helpers:validate_regex
+        extra_kwargs:
+          # Verify response parameters with a regex using positive
+          # looksaheads, so parameters match regardless of their order
+          # of appearance in the text output.
+          expression: "^\\[{{(?=.*'filename': )(?=.*'mbleft': )(?=.*'mb': )(?=.*'bytes': )(?=.*'age': )(?=.*'nzf_id': 'SABnzbd_nzf_.*')(?=.*'status': '(finished|active|queued)').*}}.*\\]$"
```

Comment:
Should we maybe leave out the tests for `text` output? Since they seem quite ugly and easy to brake..

Reply:
I agree these checks for the text version look fugly, if only because they can't be split over multiple lines and need escaping for both regex and tavern (e.g. the double `{{`). But once in place, I don't think they are particularly risky nor hard to maintain. Most of it is just a clearly delimited list of parameters being checked, easily expanded/modified/removed as needed.